imageio/core/format.py

```diff
@@ -525,10 +526,9 @@ def _set_meta_data(self, meta):
             raise NotImplementedError()
 
 
-class FormatManager(object):
+class FormatManager(metaclass=Singleton):
```

Comment:
I'm not a fan TBH :)  I avoid using metaclasses if possible. People are not supposed to instantiate this class, but use the one global instance.

Reply:
The global `imageio.formats` variable still exists. It still is the intended way of accessing the format manager from the outside.

The problem with instantiating it in the top-level `__init__` is that you can't easily reference that instance from inside the library without introducing a circular import. I think this was fine until now, but for the new API I need a reference to the singleton inside the `LegacyPlugin`. 

Having a random global inside some module of the package (what is currently the case for `imopen`) is very non-ideal; hence, the metaclass solution. It will create a new instance on the first constructor call (typically on `import imageio`), and then return the same object on each consecutive call. This way, we can do `from .core.format import FormatManager` anywhere in the codebase and use `FormatManager()` later while each time referring to the same singleton and continue to support inheritance. It's quite OOP friendly.

Here is a neat summary of various ways to do singletons in Python with their pros and cons (in case this is interesting): https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python

Reply:
I reverted it to `object`.